src/index.ts

```diff
@@ -6,41 +6,76 @@
  * MIT Licensed
  */
 
-export class Crc32 {
-  #CRC32_POLYNOMIAL = 0xedb88320;
-  #table = new Uint32Array(256).fill(0);
+class InvalidInputError extends Error {
+  constructor(public data: unknown) {
+    super("Invalid input: data must be a Uint8Array.");
+    this.name = "InvalidInputError";
+  }
+}
 
-  constructor() {
-    this.#generateTable();
+class EmptyDataArrayError extends Error {
+  constructor(public data: unknown) {
+    super("Cannot calculate CRC for an empty data array.");
+    this.name = "EmptyDataArrayError";
   }
+}
 
-  #generateTable(): void {
-    for (let i = 0; i < 256; i++) {
-      let crc = i;
-      for (let j = 0; j < 8; j++) {
-        if (crc & 1) {
-          crc = (crc >>> 1) ^ this.#CRC32_POLYNOMIAL;
-        } else {
-          crc = crc >>> 1;
-        }
+const CRC32_POLYNOMIAL = 0xedb88320;
+const TABLE = new Uint32Array(256);
+
+function generateTable(): Uint32Array {
+  for (let i = 0; i < 256; i++) {
+    let crc = i;
+    for (let j = 0; j < 8; j++) {
+      if (crc & 1) {
+        crc = (crc >>> 1) ^ CRC32_POLYNOMIAL;
+      } else {
+        crc = crc >>> 1;
       }
-      this.#table[i] = crc;
     }
+    TABLE[i] = crc;
   }
+  return TABLE;
```

Comment:
there is no need to return (and assign) the `TABLE` as this is a side-effecting function. It is ok if it returns with void. Alterhatively you can skip the function altogether and leave the initializing code as top level code that will be run whenever the module is included somewhere. There is another option: you can lazily initialize the table in the `calculateCrc` function. In the first line you can check if `TABLE` is initialized and if not run `generateTable()`. This way you don't have an unused array in memory, but the first call to `calculateCrc` will be marginally slower.